android-components/components/concept/awesomebar/src/main/java/mozilla/components/concept/awesomebar/AwesomeBar.kt

```diff
@@ -157,6 +157,11 @@ interface AwesomeBar {
          **/
         fun groupTitle(): String? = null
 
+        /**
+         * A priority number for ordering the suggestions.
+         **/
+        fun priority(): Int? = null
```

Comment:
When it comes to assigning priorities to suggestion providers, it maybe a good idea to hold an int-valued enum class instead of hardcoding int values to every single class. So returning an enum like the following (instead of an int):
``` kotlin
enum class ProviderPriority(val priority: Int) {
    BOOKMARKS(5),
    COMBINED_HISTORY(4),
    HISTORY_STORAGE(4),
    SEARCH_ACTION(1),
    SEARCH_ENGINE(0),
    SEARCH(3),
    SYNC_TABS(6)
}
```
and assigning the priority values in only one class would make its maintenance easier as we could see the priority order from a single source. The priority comparison, then, can be made via the values of enums. For the null (or default) case, I do not have the context but, adding a dummy enum such as NONE(-1) would work I guess.

Reply:
Hello @kycn , that was exactly mine and @gabrielluong's and @rocketsroger's discussion yesterday. I will be implementing it the enum way. Thank you for another confirmation! Appreciate the feedback

Reply:
@kycn Definitely agree on this and it's great to also see this brought up independent of our discussion. I added some thoughts on the Fenix discussion (see https://github.com/mozilla-mobile/fenix/pull/28024#pullrequestreview-1201297534) because I would expect the consumer of the components to manage their app's provider priority order. We basically want to avoid hardcoding the `priority` in AC.